### PR TITLE
John conroy/detail summary data tests

### PR DESCRIPTION
**Pull request by john-conroy**
Adds basic tests to `SummaryData` component for entity detail pages.

### CODE REVIEW

File: context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  expect(queryByTestId('status-svg-icon')).toBeNull();
+});
+
+test('children display when provided', () => {
+  const { getByText, getByTestId } = render(
+    <SummaryData entity_type="fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public">
+      <>
+        <div>child 1</div>
+        <div>child 2</div>
+      </>
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js:37]:
```suggestion
      <div>child 1</div>
      <div>child 2</div>
```
There is a parent, so do we need the empty tags?

File: context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { render } from 'test-utils/functions';
+import SummaryData from './SummaryData';
+
+test('json button exists and has href', () => {
+  const { getByRole } = render(
+    <SummaryData entity_type="Fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  expect(getByRole('link')).toBeInTheDocument();
+  expect(getByRole('link')).not.toBeEmptyDOMElement();
+  expect(getByRole('link')).toHaveAttribute('href', `/browse/fake/fakeuuid.json`);
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js:12]:
```suggestion
  expect(getByRole('link')).toHaveAttribute('href', `/browse/fake/fakeuuid.json`);
```
Would the error on failure be too cryptic without these?

File: context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+});
+
+test('children do not display when undefined', () => {
+  const { queryByText, queryByTestId } = render(
+    <SummaryData entity_type="fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  expect(queryByTestId('summary-data-parent')).toBeNull();
+
+  const textToTest = ['child 1', 'child 2'];
+  textToTest.forEach((text) => expect(queryByText(text)).toBeNull());
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js:54]:
```suggestion
```
Maybe there's a way to check that it's empty? There's nothing special about 'child 1'... Could just as well say 'arbitrary-string-we-haven't-mentioned-anywhere'.

Comment (john-conroy) [reply context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js:52]: @mccalluc I am checking its emptiness directly above. While there isn't anything special about 'child 1', I am checking that text is being displayed. I guess I don't understand the issue?

Comment (john-conroy) [reply context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js]: @mccalluc Sorry, I misunderstood the test on which you commented. You are correct, I removed the assertion.

**Pull request merged**